test/extended/networking/multinetpolicy.go

```diff
@@ -0,0 +1,102 @@
+package networking
+
+import (
+	"fmt"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	frameworkpod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-network][Feature:MultiNetworkPolicy]", func() {
+	var oc *exutil.CLI
+	var ns string
+
+	oc = exutil.NewCLIWithPodSecurityLevel("multinetpol-e2e", admissionapi.LevelBaseline)
+
+	f := oc.KubeFramework()
+	podName := "multinetpol-test-pod-"
+
+	g.It("should enforce a network policies on secondary network", func() {
+		var err error
+		ns = f.Namespace.Name
+
+		g.By("creating a macvlan net-attach-def")
+		nad_yaml := exutil.FixturePath("testdata", "net-attach-defs", "macvlan-nad.yml")
+		err = oc.AsAdmin().Run("create").Args("-f", nad_yaml).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("launching pod with an annotation to use the net-attach-def")
+		networksTemplate := `[
+			  {
+				"name": "macvlan1-nad",		
+				"ips": ["%s"]
+			  }
+			]`
+
+		frameworkpod.CreateExecPodOrFail(f.ClientSet, ns, podName, func(pod *v1.Pod) {
+			pod.Spec.Containers[0].Args = []string{"net", "--serve", "2.2.2.1:8889"}
+			pod.ObjectMeta.Labels = map[string]string{"pod": "a"}
+			pod.ObjectMeta.Annotations = map[string]string{
+				"k8s.v1.cni.cncf.io/networks": fmt.Sprintf(networksTemplate, "2.2.2.1/24")}
+		})
+
+		testPodB := frameworkpod.CreateExecPodOrFail(f.ClientSet, ns, podName, func(pod *v1.Pod) {
+			pod.ObjectMeta.Labels = map[string]string{"pod": "b"}
+			pod.ObjectMeta.Annotations = map[string]string{
+				"k8s.v1.cni.cncf.io/networks": fmt.Sprintf(networksTemplate, "2.2.2.2/24")}
+		})
+
+		g.By("checking podB can connnect to podA")
+		output, err := oc.AsAdmin().Run("exec").Args(testPodB.Name, "--", "curl", "--connect-timeout", "1", "2.2.2.1:8889").Output()
+		o.Expect(err).To(o.Succeed(), "cmd output: %s", output)
+
+		g.By("enabling MultiNetworkPolicies on cluster")
+		enableMultiNetworkPolicy(oc)
+		defer disablMultiNetworkPolicy(oc)
```

Comment:
You're assuming that the feature is disabled by default in all clusters where the e2e suite runs, but that doesn't seem safe. I'd check to see if it's already enabled, and only do the "enable and defer disable" if it's not.

Reply:
Sure, I added logic to store initial `useMultinetworkPolicy` state and restore it afterward